bodhi/server/scripts/skopeo_lite.py

```diff
@@ -0,0 +1,472 @@
+        self._base = 'https://{}'.format(self.registry)
+        if insecure:
+            # In the insecure case, if the registry is just a hostname:port, we
+            # don't know whether to talk HTTPS or HTTP to it, so we try first
+            # with https then fallback
+            self._fallback = 'http://{}'.format(self.registry)
+
+        self.session = requests.Session()
+
+    def _find_cert_dir(self):
+        hostport = self.registry
+
+        for d in ('/etc/containers/certs.d', '/etc/docker/certs.d'):
+            certs_dir = os.path.join(d, hostport)
+            if os.path.isdir(certs_dir):
+                return certs_dir
+
+        return None
+
+    def _find_cert(self, cert_dir):
+        if cert_dir is None:
+            cert_dir = self._find_cert_dir()
+
+        if cert_dir is None:
+            return None
+
+        for l in sorted(os.listdir(cert_dir)):
+            if l.endswith('.cert'):
+                certpath = os.path.join(cert_dir, l)
+                keypath = certpath[:-5] + '.key'
+                if not os.path.exists(keypath):
+                    raise RuntimeError("Cannot find key file for {}".format(certpath))
+                return (certpath, keypath)
+            elif l.endswith('.key'):
+                # Should have found <x>.cert first
+                keypath = os.path.join(cert_dir, l)
+                raise RuntimeError("Cannot find certificate file for {}".format(keypath))
+
+        return None
+
+    def _do(self, f, relative_url, *args, **kwargs):
+        kwargs['auth'] = self.auth
+        kwargs['cert'] = self.cert
+        kwargs['verify'] = not self.insecure
+        res = None
+        if self._fallback:
+            try:
+                res = f(self._base + relative_url, *args, **kwargs)
+                self._fallback = None  # don't fallback after one success
+            except (SSLError, ConnectionError):
+                self._base = self._fallback
+                self._fallback = None
+        if res is None:
+            res = f(self._base + relative_url, *args, **kwargs)
+        return res
+
+    def get(self, relative_url, data=None, **kwargs):
+        return self._do(self.session.get, relative_url, **kwargs)
+
+    def head(self, relative_url, data=None, **kwargs):
+        return self._do(self.session.head, relative_url, **kwargs)
+
+    def post(self, relative_url, data=None, **kwargs):
+        return self._do(self.session.post, relative_url, data=data, **kwargs)
+
+    def put(self, relative_url, data=None, **kwargs):
+        return self._do(self.session.put, relative_url, data=data, **kwargs)
+
+
+class ManifestInfo(object):
+    def __init__(self, contents, digest, media_type, size):
+        self.contents = contents
+        self.digest = digest
+        self.media_type = media_type
+        self.size = size
+
+
+def get_manifest(session, repository, ref):
+    """Download a manifest from a registry. ref can be a digest, or a tag."""
```

Comment:
We need to document each of the parameters and their types, and the return value and its type.

Reply:
For docblocks, Bodhi uses this style:

```
"""
Do a thing.

Args:
    x (bool): blah blah.
    y (int): Neat other stuff.
Returns:
    str: A string of cool thoughts.
Raises:
    ValueError: If you forgot to tie your shoes.
"""
```